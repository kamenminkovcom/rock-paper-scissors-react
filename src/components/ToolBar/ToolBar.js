import React from 'react';
import HandButton from './HandButton/HandButton';
import PropTypes from 'prop-types';

const ToolBar = ({handTypes, choseHandType}) => (
    <div className="tool-bar">
        {Object.values(handTypes).map(h => (
            <HandButton handType={h}
                        handleClick={() => choseHandType(h)}
                        key={h}/>
        ))}
    </div>
);

ToolBar.propTypes = {
    handTypes: PropTypes.object.isRequired,
    choseHandType: PropTypes.func.isRequired
};

export default ToolBar;