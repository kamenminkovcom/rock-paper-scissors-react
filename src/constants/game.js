export const handTypes = {
    paper: 'paper',
    rock: 'rock',
    scissors: 'scissors',
};

export const gameRules = {
    [handTypes.paper]: {
        [handTypes.paper]: 0,
        [handTypes.rock]: 1,
        [handTypes.scissors]: 0,
    },
    [handTypes.rock]: {
        [handTypes.rock]: 0,
        [handTypes.scissors]: 1,
        [handTypes.paper]: 0
    },
    [handTypes.scissors]: {
        [handTypes.scissors]: 0,
        [handTypes.paper]: 1,
        [handTypes.rock]: 0
    }
};

export const possibleResults = {
    win: 'win',
    lose: 'lose',
    draw: 'draw'
};

//may we would like to change classNames and it should not break the logic
export const classNames = {
    [possibleResults.win]: 'win',
    [possibleResults.lose]: 'lose',
    [possibleResults.draw]: 'draw'
};

export const gameHistoryCount = 5;

export const defaultChoices = {
    cpuChoice: undefined,
    playerChoice: undefined
};