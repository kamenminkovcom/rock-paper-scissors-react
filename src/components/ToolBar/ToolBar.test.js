import React from 'react';
import {shallow} from 'enzyme';
import ToolBar from './ToolBar';
import {handTypes} from '../../constants/game';


describe('<ToolBar/>', () => {
    it('Should render correctly', () => {
        const wrapper = shallow(<ToolBar handTypes={Object.values(handTypes)} choseHandType={() => {}}/>);
        expect(wrapper).toMatchSnapshot();
    });
});