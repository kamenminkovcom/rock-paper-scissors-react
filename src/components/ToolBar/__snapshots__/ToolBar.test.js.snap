// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolBar/> Should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToolBar
    choseHandType={[Function]}
    handTypes={
        Array [
            "paper",
            "rock",
            "scissors",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HandButton
          handType="0"
          handleClick={[Function]}
/>,
        <HandButton
          handType="1"
          handleClick={[Function]}
/>,
        <HandButton
          handType="2"
          handleClick={[Function]}
/>,
      ],
      "className": "tool-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "handType": "0",
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "handType": "1",
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "handType": "2",
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HandButton
            handType="0"
            handleClick={[Function]}
/>,
          <HandButton
            handType="1"
            handleClick={[Function]}
/>,
          <HandButton
            handType="2"
            handleClick={[Function]}
/>,
        ],
        "className": "tool-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "handType": "0",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "handType": "1",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "handType": "2",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
